Source/WebCore/page/EventHandler.cpp

```diff
@@ -4337,49 +4337,105 @@ void EventHandler::defaultBackspaceEventHandler(KeyboardEvent& event)
         event.setDefaultHandled();
 }
 
-KeyboardScrollingAnimator* EventHandler::keyboardScrollingAnimatorForFocusedNode()
+void EventHandler::stopKeyboardScrolling()
 {
-    Node* node = m_frame.document()->focusedElement();
+    auto* animator = m_frame.page()->currentKeyboardScrollingAnimator();
+    if (animator)
+        animator->handleKeyUpEvent();
+}
 
-    if (!node)
-        node = m_mousePressNode.get();
+bool EventHandler::beginKeyboardScrollGesture(KeyboardScrollingAnimator* animator, KeyboardEvent& event)
+{
+    auto* platformEvent = event.underlyingPlatformEvent();
+
+    if (animator && platformEvent)
+        return animator->beginKeyboardScrollGesture(*platformEvent, m_frame.page());
+
+    return false;
+}
+
+bool EventHandler::startKeyboardScrollingOnFrame(KeyboardEvent& event)
```

Comment:
"OnFrame" sounds like this would take a `Frame&` argument, but it means "this EventHanders's Frame", so might be better named `startKeyboardScrollingOnDocument`.